dashboard/src/components/AppView/AppView.tsx

```diff
@@ -195,23 +192,27 @@ export default function AppView() {
         helm={true}
         icon={icon}
         buttons={[
-          <Link to={url.app.apps.upgrade(cluster, namespace, releaseName)} key="upgrade-button">
-            <CdsButton status="primary">
-              <CdsIcon shape="upload-cloud" /> Upgrade
-            </CdsButton>
-          </Link>,
+          <UpgradeButton
```

Comment:
Thanks for adding the RollbackButton component. I don't know if, at this moment, it worths just having the `StatusAwareButton` with an additional property `type="Upgrade|Rollback|Delete". This way we would be DRYer no?

Just an idea, I'm totally fine leaving this PR as is

Reply:
I am not familiar with what you mean by "DRYer"?
the StatusAwareButton only deals with disabling and adding a tooltip. Adding a type would inject more state than its needs to have.
what would be the benefit of adding type to it?

Reply:
Hah - too many acronyms sometimes. I meant DRY as in https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

I just felt we are replicating too much code in each `UpgradeButton`, `RollbackButton` and `DeleteButton` (eg. unit tests). Perhaps it's better to remain them as separated components as a single "generic" status aware button would deal with too much logic and state.
Just wanted to open up the discussion, but I'm ok with the current implementation!